tests/themispp/secure_session_test.hpp

```diff
@@ -27,104 +27,134 @@
 
 #include <themispp/secure_session.hpp>
 
+#include "utils.hpp"
+
 // Allow usage of non-owning Secure Session constructor for testing
 #if defined(__GNUC__) || defined(__clang__)
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wdeprecated-declarations"
 #endif
 
+// Thank you, STL, for providing this out-of-the-box. It's really convenient.
```

Comment:
where we use it?

Reply:
std::unordered_map used in secure_session_moved() test requires the key type to be hashable. We use std::vectors as keys and they are not hashable, unfortunately.

Reply:
actually I don't like this comment because we look like kids that crying due to needing write a small piece of code. especially when STL provides a way to extend base structures with custom types and vector which supports a lot of other types may be implemented in different ways. but it is up to you leave or remove :)

Reply:
Yes, it does piss me off personally a bit because every other language that I known which has built-in 'vectors' and 'hash maps' allows to use vectors as map keys right away, without having to specialize _anything_ in their standard library.

I'm not even 100% sure that it's legal to do it the way I do here. I remember that one is not allowed to _overload_ any function in namespace std, but it's okay to provide specializations for templates there. However, I'm not sure that it's okay to provide specialization for _all_ vectors – maybe std::vector<bool> is special again.

Since you've brought this up I guess I'll put my serious, professional hat on and replace this comment with something more neutral.